### DISCUSSION TITLE
How to be part of Scoop maintainer team?

### DISCUSSION THREAD

**Discussion by jasongodev** [ScoopInstaller/Extras#9981] [General]
I like Scoop and used it for almost everything. I'm a developer and it makes tooling easy on Windows.

I want to dedicate a portion of my time, up to 1 hour every day, to check, test, and fast track some pull requests in the Scoop buckets repositories (e.g. main and extras). I can see some of the requests are very easy to merge, mostly are url updates of the installers, new version number sources, etc.

If I can test and bless the pull requests for merging then we can reduce the number of pull requests at a regular pace. Slowly, surely, but at least on regular pace.

Let me know your thoughts.

**Comment by niheaven** [comment]
You could start by reviewing the PRs and leaving your comments, everyone in the maintainer team does that first.

**Comment by benf-write** [comment]
I'm beginning to feel a bit concerned Scoop's maintenance because there are many PRs waiting on reviews. I understand that it's important to be judicious about what gets accepted, but it seems like there have not been many active reviewers recently. If Jason could help with that, it would be great.

**Comment by pureby** [comment]
160+ PR's, and the guy who wants to help an hour a day for free is denied the opportunity?

I don't understand how Scoop wants to grow like this. If developers can't add their own apps to Scoop, it can get very frustrating and they might lose interest as result.

**Reply by jasongodev** [reply]
I believe there is real opportunity to do this and it was mentioned in previous comments that I can start commenting and testing some of the pull requests to get going. That's enough go signal that this project is accepting help.

My Windows time is significantly lower these days as I am using Arch Linux now as my daily driver. But I still maintain another laptop that has Windows in it. So I will test some things here and there from time to time.

**Reply by pureby** [reply]
Thanks for clarification; I might have misunderstood the reply from @niheaven to you. 

In any case, I'm very happy to see people like you who have the knowledge and ability to help this amazing project. I absolutely love Scoop, but am a bit concerned about it outgrowing it's current maintainer capacity, since the number of unprocessed PR's is steadily increasing over the last months.

**Reply by niheaven** [reply]
I must clear again our attitude towards new maintainers: don't just SAY, DO it.

We've added some new maintainers recently and they all create and review lots of issues and PRs. We do part-time jobs on Scoop and may not have enough time to review issues and PRs in time, so maintainers are always welcomed. But users should show us they have some good manners and abilities to do this job, and this ensures a healthy community.

**Reply by pureby** [reply]
> I must clear again our attitude towards new maintainers: don't just SAY, DO it.
> 
> We've added some new maintainers recently and they all create and review lots of issues and PRs. We do part-time jobs on Scoop and may not have enough time to review issues and PRs in time, so maintainers are always welcomed. But users should show us they have some good manners and abilities to do this job, and this ensures a healthy community.

Thanks for the clarification. 

Showing good manners speaks for itself, of course, but _"showing the ability to do the job"_ is very hard if you're not a developer yourself. I would like to contribute some of my free time to the project by testing new manifests for example, but since I'm not a developer and there is no accessible documentation on how to build/test manifests, my hands are tied.

Maybe improving the documentation and targeting it at non-developers can gain Scoop more support from the user-base in return? What do you think?

**Reply by rasa** [reply]
Yes, documentation for our maintainers is a good idea. I have been using Windows Sandbox for testing manifests using this script [this script](https://github.com/ScoopInstaller/Scoop/pull/5349) and once it's merged I will update our wiki.

**Comment by jasongodev** [comment]
I have started to review a couple of pull requests this week.

You can view my reviews here: https://github.com/ScoopInstaller/Extras/pulls?q=is%3Apr+is%3Aopen+reviewed-by%3Ajasongodev

My reviews involves testing the manifest file in a real Windows environment. I test installation, uninstall, and autoupdate feature (using checkver.ps1). I will not add/edit the manifest files or pull requests and I will usually just ask the PR's author to do the changes themselves as suggested.

Basically I'm the second eyes for the automated github-actions.

I will also prioritize PRs that are safe to merge without further discussions so we can decrease the 230+ PRs to date.

I can only promise an average of 2 reviews per day. Some days can come in bulk, some days none. I hope it helps.
